### PR TITLE
Wallet: Add address to watch

### LINKED ISSUE #17469: Wallet: Add address to watch screen
Implement the "add address to watch" screen.

[Designs](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=2052-522319&mode=design&t=PJiI0P0LEHGBakdp-0)

This issue is for UI implementation of the screen

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/17469

This PR the implements "Add address to watch" screen UI.

[Designs](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=2052-522319&mode=design&t=6x39EB1JVkCXniFS-0)

Demo:

https://github.com/status-im/status-mobile/assets/29354102/45710fbc-e610-4850-840f-1ebb93c859cd

### CODE REVIEW

File: src/status_im2/contexts/wallet/address_watch/view.cljs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+       [quo/text-combinations
+        {:container-style style/header-container
+         :title           (i18n/label :t/add-address)
+         :description     (i18n/label :t/enter-eth)}]
+       [rn/view {:style style/input-container}
+        [quo/input
+         {:label           (i18n/label :t/eth-or-ens)
+          :button          {:on-press (fn [] (clipboard/get-string #(reset! input-value %)))
+                            :text     (i18n/label :t/paste)}
+          :placeholder     (str "0x123abc... " (string/lower-case (i18n/label :t/or)) " bob.eth")
```
Comment (J-Son89) [inline src/status_im2/contexts/wallet/address_watch/view.cljs:36]: Should this whole line not be a translation string? 🤔

Comment (OmarBasem) [reply src/status_im2/contexts/wallet/address_watch/view.cljs:36]: @J-Son89 I don't think so. That's an Ethereum address and ENS name.

Comment (J-Son89) [reply src/status_im2/contexts/wallet/address_watch/view.cljs:36]: Yeah good point. I would still say we're assuming how other languages would construct this but should be fine.

**Review by J-Son89 — APPROVED** [review]
Nice one @OmarBasem! 🙌🏼

**Review by ibrkhalil — APPROVED** [review]

File: src/status_im2/contexts/wallet/address_watch/view.cljs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+       [quo/page-nav
+        {:type      :no-title
+         :icon-name :i/close
+         :on-press  #(rf/dispatch [:navigate-back])}]
+       [quo/text-combinations
+        {:container-style style/header-container
+         :title           (i18n/label :t/add-address)
+         :description     (i18n/label :t/enter-eth)}]
+       [rn/view {:style style/input-container}
+        [quo/input
```
Comment (briansztamfater) [inline src/status_im2/contexts/wallet/address_watch/view.cljs:32]: We should use monospace font when dealing with crypto addresses

Comment (briansztamfater) [reply src/status_im2/contexts/wallet/address_watch/view.cljs:32]: I see Figma uses Regular font in this case but seems suspicious to me. Maybe double check with design team cc @Francesca-G @pedro-et 🤔

Comment (Francesca-G) [reply src/status_im2/contexts/wallet/address_watch/view.cljs:32]: @briansztamfater as confirmed by Pedro [here](https://discord.com/channels/624634427930312714/852533718790570015/1159061770078539817) everytime it has an address it should be monospsace :)

**Review by vkjr — APPROVED** [review]

**Pull request merged**